test/unit/api/api-record-log-events.test.js

```diff
@@ -0,0 +1,173 @@
+/*
+ * Copyright 2022 New Relic Corporation. All rights reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+'use strict'
+
+const tap = require('tap')
+const helper = require('../../lib/agent_helper.js')
+const API = require('../../../api.js')
+const { SUPPORTABILITY, LOGGING } = require('../../../lib/metrics/names')
+const API_METRIC = SUPPORTABILITY.API + '/recordLogEvent'
+
+tap.test('Agent API - recordCustomEvent', (t) => {
+  t.autoend()
+
+  let agent = null
+  let api = null
+  const message = 'just logging a log in the logger'
+
+  t.beforeEach(() => {
+    agent = helper.loadMockedAgent()
+    api = new API(agent)
+  })
+
+  t.afterEach(() => {
+    helper.unloadAgent(agent)
+    agent = null
+    api = null
+  })
+
+  t.test('can handle a singular log message', (t) => {
+    const now = Date.now()
+    t.doesNotThrow(() => {
+      api.recordLogEvent({ message })
+    }, 'does not throw when logging a single message')
+
+    const logMessage = popTopLogMessage(agent)
+    t.ok(logMessage, 'we have a log message')
+    t.equal(logMessage.message, message, 'it has the right log message')
+    t.equal(logMessage.level, 'UNKNOWN', 'it has UNKNOWN severity')
+    t.ok(logMessage.timestamp >= now, 'its timestamp is current')
+    t.ok(logMessage.hostname, 'a hostname was set')
+    t.notOk(logMessage['trace.id'], 'it does not have a trace id')
+    t.notOk(logMessage['span.id'], 'it does not have a span id')
+
+    const lineMetric = agent.metrics.getMetric(LOGGING.LINES)
+    t.ok(lineMetric, 'line logging metric exists')
+    t.equal(lineMetric.callCount, 1, 'ensure a single log line was counted')
+
+    const unknownLevelMetric = agent.metrics.getMetric(LOGGING.LEVELS.UNKNOWN)
+    t.ok(unknownLevelMetric, 'unknown level logging metric exists')
+    t.equal(unknownLevelMetric.callCount, 1, 'ensure a single log line was counted')
+
+    const apiMetric = agent.metrics.getMetric(API_METRIC)
+    t.ok(apiMetric, 'API logging metric exists')
+    t.equal(apiMetric.callCount, 1, 'ensure one API call was counted')
+
+    t.end()
+  })
+
+  t.test('adds the proper linking data in a transaction', (t) => {
+    const now = Date.now()
+
+    helper.runInTransaction(agent, 'logging-api-test', (tx) => {
+      api.recordLogEvent({ message, level: 'info' })
+      tx.end()
+    })
+    const logMessage = popTopLogMessage(agent)
+
+    t.ok(logMessage, 'we have a log message')
+    t.equal(logMessage.message, message, 'it has the right log message')
+    t.equal(logMessage.level, 'info', 'it has `info` severity')
+    t.ok(logMessage.timestamp >= now, 'its timestamp is current')
```

Comment:
this is fine but you could use sinon.useFakeTimers()

Reply:
Leaving this one as-is, copied this idiom from existing tests.